### ISSUE TITLE
Error while running segmentation model

### ISSUE THREAD

**Issue by sameerkhann** [NVIDIA/DIGITS#899]
Hi i was trying to implement the segmentation model according to this
https://github.com/NVIDIA/DIGITS/pull/830/commits/caccc9d94dc968283a58567b6ae77482dcfdb7d4
But when i clicked the create button this error pops up:
Creating layer lossL1
Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: L1Loss (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, Concat, ContrastiveLoss, Convolution, Data, Deconvolution, Dropout, DummyData, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Pooling, Power, Python, ReLU, Reduction, Reshape, SPP, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, WindowData)

**Comment by gheinrich** [comment]
Hello, to use the Caffe model you need [nv-caffe 0.15](https://github.com/NVIDIA/caffe/tree/caffe-0.15). Or you can use `EuclideanLoss` however you'll need to set the learning rate to a very small value.

**Comment by sameerkhann** [comment]
i am using nv-caffe 0.15, i tried the SoftmaxWithLoss the training error and validation error is like this:
![untitled](https://cloud.githubusercontent.com/assets/15321510/16759760/148f3632-484b-11e6-9372-708c36e1a5fe.png). 
i think it is not learning well enough.

**Comment by sameerkhann** [comment]
Also when i am trying to run this segmentation model on my own brain data
it gives me this error:

Check failed: bottom[0]->count(1) == bottom[1]->count(1) (24576 vs.
81408) Inputs must have the same dimension.

**Comment by sameerkhann** [comment]
My task is to do the semantic segmentation of the skull mask and brain part.  i am confused what will be my label folder and what will be my train folder. herewith attached are my example images
![wholebrain](https://cloud.githubusercontent.com/assets/15321510/16763799/30407ece-485d-11e6-88ed-e76757a830fa.jpg)
![img148](https://cloud.githubusercontent.com/assets/15321510/16763810/3e4eb26a-485d-11e6-85ae-f3b5df111e55.jpeg)
![img148](https://cloud.githubusercontent.com/assets/15321510/16763821/4d7888a6-485d-11e6-8711-88d881b50911.jpeg)
kindly help me in figuring it out.
Best regards

**Comment by gheinrich** [comment]
Caffe 0.15 comes with a new [L1 loss layer](https://github.com/NVIDIA/caffe/blob/caffe-0.15/src/caffe/layers/l1_loss_layer.cpp) so you shouldn't get this error. You might want to check that you are pointing to the right location for Caffe (`./digits-devserver -c` then specify the path to Caffe explicitly). Alternatively as I said you can use `EuclideanLoss` (and reduce the learning rate if the network does not converge). `SoftmaxWithLoss` is irrelevant for this kind of problem.

The idea behind the very simple FCN that I provide in the example is to compare the output of the network with the label image. The source image is the image you want to segment. The target image is the segmented image to use for training. In my example, those have the same dimensions, though that isn't required if you update the model accordingly.

For your specific problem you might need a deeper network than the one I provide in the example. Have a look at this [paper](http://people.eecs.berkeley.edu/~jonlong/long_shelhamer_fcn.pdf) for more background information.

**Comment by gheinrich** [comment]
No feedback => closing.
